### DISCUSSION TITLE
How do I extend the window object?

### DISCUSSION THREAD

**Discussion by CodyBontecou** [vercel/next.js#28604] [Help]
I'm working on getting [mswjs](https://mswjs.io/) to work alongside my [cypress](https://www.cypress.io/) tests and was recommended [https://mswjs.io/docs/api/setup-worker/use#examples](https://mswjs.io/docs/api/setup-worker/use#examples).

The code example suggests I expose msw to the window object like so:

```js
import { setupWorker, rest } from 'msw'

const worker = setupWorker(
  ...
)

window.msw = {
  worker,
  rest,
}
```
But.. I'm not entirely sure how to do that in a Nextjs project. Any pointers?

**Comment by delbaoliveira** [comment]
Hey @CodyBontecou, 

Are you getting any errors while trying to work with the window object? 

For pointers, there's a [Next.js with MSW example](https://github.com/vercel/next.js/tree/canary/examples/with-msw/mocks) where [window is inside a conditional](https://github.com/vercel/next.js/blob/canary/examples/with-msw/mocks/index.js) - this is to prevent Node.js errors when trying to run the _window_ object server-side.

**Reply by CodyBontecou** [reply]
Hey @delbaoliveira 

Thank you for the response. I've tried attaching some msw-specifics to the window object like so:

```js
import { rest } from 'msw'

if (typeof window === 'undefined') {
  const { server } = require('./server')
  server.listen()
} else {
  const { worker } = require('./browser')
  worker.start()
  window.msw = {
    worker,
    rest,
  }
}
```

but it's not giving me the ability to `console.log(window.msw)` within my cypress tests.
